WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift

```diff
@@ -162,6 +162,15 @@ final class StorePerformanceViewModel: ObservableObject {
 
         // Stop if data is relatively new
         if !forceRefresh && DashboardTimestampStore.isTimestampFresh(for: .performance, at: timeRange.timestampRange) {
+
+            // We can show the cached data since we know it's available from the timestamp.
+            switch timeRange {
+            case .custom:
+                updateSiteVisitStatModeForCustomRange()
+            case .today, .thisWeek, .thisMonth, .thisYear:
+                siteVisitStatMode = .default
+            }
```

Comment:
❓ To avoid duplicating code, could we move the logic from [these lines](https://github.com/woocommerce/woocommerce-ios/blob/c297a96f51a2274202e56905d43dbc7003d465c5/WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift#L188-L197) into a helper method and use that?

As an added advantage I see that the original login updates the widget in `today` case.

Reply:
I updated it in https://github.com/woocommerce/woocommerce-ios/pull/14438/commits/3a02b98c10ad668ea5c7c17e836a6d3cbf6905be

I put the Widget updating outside of the helper function, though, since it has nothing to do with `siteVisitStatMode` that the function is handling.